src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -51,6 +52,7 @@ public CircuitHost(
         RemoteNavigationManager navigationManager,
         CircuitHandler[] circuitHandlers,
         CircuitMetrics? circuitMetrics,
+        ComponentsActivitySource? componentsActivitySource,
```

Comment:
The activity source and the metrics shouldn't be nullable, they should be the same as RemoteJsRuntime and other parameters.

We unconditionally register them https://github.com/dotnet/aspnetcore/pull/61609/files#diff-e82512017622b574554e1a1519f2bb3bbb1d8957e5346b3a4cbeb0f10e368c79R64 and

https://github.com/dotnet/aspnetcore/pull/61609/files#diff-9e461d4480403d04ed6d5e6c31c44261fe8d5926969ab338dd910564ea211396R79-R80

We should throw an exception if they aren't being provided.